### DISCUSSION TITLE
Autologin with LDAP authentication

### DISCUSSION THREAD

**Discussion by mecrider** [ltsp/ltsp#312] [Q&A] [answered]
I am running LTSP 20.06 on Ubuntu 20.04 with a chrootless image. I configured LDAP authentication on the server with NFS home directories and all clients are able to log in without issue with any LDAP username and password. However, if I attempt to set up autologin for a particular client I end up at the GDM login screen after bootup. /etc/gdm3/custom.conf has the proper entries in a [daemon] section at the bottom, but since the user is in LDAP it does not exist in /etc/shadow for 54-pam.sh to update. Is there a way to configure autologin so it works with LDAP users?

**Comment by alkisg** [comment]
LDAP support in LTSP is tracked in https://github.com/ltsp/ltsp/issues/16.
For workarounds before properly implementing LDAP, let's use community discussion issues instead.

I suppose that you used OMIT_FUNCTIONS="pam_main" to skip pam, since you manually configured LDAP. So 54-pam.sh shouldn't be executed at all.

AUTOLOGIN is configured in 55-display-manager.sh, are there any issues there?

**Comment by mecrider** [comment]
I'm finally looking back at this, now running LTSP 21.01-1 on Ubuntu 20.04. Based on issue 338 I have added 'PAM_AUTH_TYPE="0"' under [clients] in ltsp.conf, and logins are flawless. However, where setting AUTOLOGIN in ltsp.conf formerly resulted in proper entries in gdm3/custom.conf that seemed to be ignored by GDM, now ltsp is adding "AutomaticLoginEnable=false" and "AutomaticLogin=" no matter what LDAP username I put in the AUTOLOGIN parameter. I have verified that I can log in as root to a virtual terminal and run "getent passwd" on the user and get results. If I put the lines in a separate file and load it with GDM3_CONF the user is logged in.

**Comment by alkisg** [comment, marked as answer]
Put a `debug_shell` at the end of 55-display-manager.sh > display_manager_main(), run `ltsp initrd` to put the changes in ltsp.img, reboot a client, and in that shell, run:

```shell
echo "HOSTNAME=$HOSTNAME"
echo "AUTOLOGIN=$AUTOLOGIN"
echo "_AUTOLOGIN=$_AUTOLOGIN"
echo "_ALUSER=$_ALUSER"
getent passwd "$AUTOLOGIN"
cat /etc/gdm3/custom.conf
```
Then paste the results here. I'm guessing that at that point, LDAP doesn't work yet.

If that's the case, try to remove the `getent passwd` check in that function, and see if things work properly.
If they do, please file a bug report about this, so that we change it to `if LDAP ... then don't use getent there...`.

Btw, note that a workaround should be to put the following in ltsp.conf:

```shell
GDM3_CONF="AutomaticLoginEnable=true
AutomaticLogin=user"
```

**Comment by mecrider** [comment]
HOSTNAME=logintest
AUTOLOGIN=losscc
_AUTOLOGIN=false
_ALUSER=
no response on "getent passwd"
appended section of custom.conf:
[daemon]
AutomaticLoginEnable=false
AutomaticLogin=

Your guess is correct. Removing the getent passwd check makes it work properly. I have opened issue #399 on this.

The workaround is what I was using before your reply.

Thanks for your help.
